epochStart/shardchain/trigger.go

```diff
@@ -443,14 +463,30 @@ func (t *trigger) getHeaderWithNonceAndHash(nonce uint64, neededHash []byte) (*b
 
 // call only if mutex is locked before
 func (t *trigger) getHeaderWithNonceAndPrevHashFromMaps(nonce uint64, prevHash []byte) *block.MetaBlock {
+	lowestRound := uint64(math.MaxUint64)
+	chosenMeta := &block.MetaBlock{}
+
 	metaHdrHashesWithNonce := t.mapNonceHashes[nonce]
 	for _, hash := range metaHdrHashesWithNonce {
 		hdrWithNonce := t.mapHashHdr[hash]
-		if hdrWithNonce != nil && bytes.Equal(hdrWithNonce.PrevHash, prevHash) {
-			return hdrWithNonce
+		if check.IfNil(hdrWithNonce) {
+			continue
+		}
+		if !bytes.Equal(hdrWithNonce.PrevHash, prevHash) {
+			continue
+		}
+
+		if lowestRound > hdrWithNonce.GetRound() {
+			lowestRound = hdrWithNonce.GetRound()
+			chosenMeta = hdrWithNonce
 		}
 	}
-	return nil
+
+	if chosenMeta.GetNonce() != nonce {
```

Comment:
You can check: if lowestRound == uint64(math.MaxUint64) instead this one, as there could be an edge case when you call this method with parameter nonce = 0 and you don't find anything

Reply:
I know, but with this check you will cover any situation